zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/schema/AbstractSchema.scala

```diff
@@ -199,26 +199,26 @@ abstract class Schema(typeName: String,
 
   def getTypeName: String = typeName
   def getTimeField: TimeField
-  def copySchema: Schema
-  def asTemporal: TemporalSchema
-  def isTemporal: Boolean
+  def copySchema: AbstractSchema
+  def asSchema: Schema
```

Comment:
I still feel `asSchema` should not be a method of the base class. if needed, we can always convert it using `asInstanceOf`